src/main/java/com/redlimerl/speedrunigt/events/Event.java

```diff
@@ -35,26 +38,17 @@ public Event(
         try {
             String[] parts = eventString.trim().split(" ");
 
-            String eventId;
-            Integer version = 0;
-            String type;
-            int offset;
-            try {
-                eventId = parts[0];
-                type = parts[offset = 1];
-            } catch (Exception ignored) {
-                eventId = parts[offset = 0];
-                version = runningVersions.getOrDefault(eventId, 0);
-                type = "common";
+            int wordPointer = 0;
+            String eventId = parts[wordPointer++];
+            String type = eventId.split("\\.")[1];
+            long realTime = Long.parseLong(parts[wordPointer++]);
+            long gameTime = Long.parseLong(parts[wordPointer++]);
+            int version = runningVersions.getOrDefault(eventId, 0);
+            if (wordPointer < parts.length) {
+                version = Integer.parseInt(parts[wordPointer]);
             }
 
-            Long realTime = Long.parseLong(parts[offset++]);
-            Long gameTime = Long.parseLong(parts[offset++]);
-            if (parts.length >= offset + 1) {
-                version = Integer.parseInt(parts[offset + 1]);
-            }
-
-            return new Event(version, eventId, type, realTime, gameTime);
+            return new Event(version, eventId, type, EventFactoryLoader.isEventRepeatable(eventId, type), realTime, gameTime);
```

Comment:
Not a fan of having to call this static isEventRepeatable method to determine if it is repeatable here, but it isn't stored in the events.log so idk how I'd get it otherwise